integrations-and-supported-tools/detectron2/scripts/Neptune_detectron2.py

```diff
@@ -0,0 +1,145 @@
+from detectron2.engine import DefaultPredictor, DefaultTrainer
+from detectron2.structures import BoxMode
+from detectron2.utils.logger import setup_logger
+from detectron2.utils.visualizer import ColorMode, Visualizer
+from neptune_detectron2 import NeptuneHook
+
+setup_logger()
+
+# (Neptune) Initialize a new run
+run = neptune.init_run(
+    api_token=neptune.ANONYMOUS_API_TOKEN,
+    project="common/detectron2-integration",
+)
+
+# ### Transform dataset for training
+
+
+def get_balloon_dicts(img_dir):
+    json_file = os.path.join(img_dir, "via_region_data.json")
+    with open(json_file) as f:
+        imgs_anns = json.load(f)
+
+    dataset_dicts = []
+    for idx, v in enumerate(imgs_anns.values()):
+        record = {}
+
+        filename = os.path.join(img_dir, v["filename"])
+        height, width = cv2.imread(filename).shape[:2]
+
+        record["file_name"] = filename
+        record["image_id"] = idx
+        record["height"] = height
+        record["width"] = width
+
+        annos = v["regions"]
+        objs = []
+        for _, anno in annos.items():
+            assert not anno["region_attributes"]
+            anno = anno["shape_attributes"]
+            px = anno["all_points_x"]
+            py = anno["all_points_y"]
+            poly = [(x + 0.5, y + 0.5) for x, y in zip(px, py)]
+            poly = [p for x in poly for p in x]
+
+            obj = {
+                "bbox": [np.min(px), np.min(py), np.max(px), np.max(py)],
+                "bbox_mode": BoxMode.XYXY_ABS,
+                "segmentation": [poly],
+                "category_id": 0,
+            }
+            objs.append(obj)
+        record["annotations"] = objs
+        dataset_dicts.append(record)
+    return dataset_dicts
+
+
+for d in ["train", "val"]:
+    DatasetCatalog.register("balloon_" + d, lambda d=d: get_balloon_dicts("balloon/" + d))
+    MetadataCatalog.get("balloon_" + d).set(thing_classes=["balloon"])
+balloon_metadata = MetadataCatalog.get("balloon_train")
+
+
+# Configure the model for training
+device = "cuda" if torch.cuda.is_available() else "cpu"
+
+cfg = get_cfg()
+cfg.MODEL.DEVICE = device
+cfg.DATALOADER.NUM_WORKERS = 0
+cfg.merge_from_file(
+    model_zoo.get_config_file("COCO-InstanceSegmentation/mask_rcnn_R_50_FPN_3x.yaml")
+)
+cfg.DATASETS.TRAIN = ("balloon_train",)
+cfg.DATASETS.TEST = ()
+cfg.MODEL.WEIGHTS = model_zoo.get_checkpoint_url(
+    "COCO-InstanceSegmentation/mask_rcnn_R_50_FPN_3x.yaml"
+)
+cfg.SOLVER.IMS_PER_BATCH = 2
+cfg.SOLVER.BASE_LR = 0.00025
+cfg.SOLVER.MAX_ITER = 30
+cfg.SOLVER.STEPS = []
+cfg.MODEL.ROI_HEADS.BATCH_SIZE_PER_IMAGE = 128
+# only has one class (balloon)
+# see https://detectron2.readthedocs.io/tutorials/datasets.html#update-the-config-for-new-datasets
+cfg.MODEL.ROI_HEADS.NUM_CLASSES = 1
+
+
+# (Neptune) Create a hook using Neptune integration
+# NOTE: You can also log checkpoints by passing `log_checkpoints=True`.
+hook = NeptuneHook(run=run, log_model=True, metrics_update_freq=10)
```

Comment:
Have opted out of logging checkpoints as each checkpoint is around 330MB. Instead have added a comment of how this can be done.